Add support for single category layer legend items (ref: missing category and title)

volterra79 opened the pull request:
Closes: #377

volterra79 committed:
Categories need to be more than 0

Raruto commented on src/app/core/layers/geolayermixin.js:
```diff
@@ -108,7 +108,7 @@ proto.setCategories = function(categories=[]) {
   this.legendCategories[this.getCurrentStyle().name] = categories;
   //set categories state attribute to true only if exist at least a rule key
   // meaning that layer has at least more than one has a
-  this.state.categories = categories.length > 1 && categories.filter(category => category.ruleKey).length > 1;
+  this.state.categories = categories.length > 0 && categories.filter(category => category.ruleKey).length > 0;
```
double check the comment above

volterra79 replied:
@Raruto done

volterra79 committed:
Merge branch 'dev' into fix_ux_single_categoriey

volterra79 committed:
Change comment

Raruto commented:
@volterra79 for me we can merge it (I haven't tested, but it seems quite reasonable).

BTW, there was a particular reason why before the `length` check was set to greater than one?

volterra79 committed:
Add checked and ruleKey on case of just one category is set

volterra79 commented:
> @volterra79 for me we can merge it (I haven't tested, but it seems quite reasonable).
> 
> BTW, there was a particular reason why before the `length` check was set to greater than one?

@Raruto it was related to (i don't remember) old way to send categories from a server

volterra79 committed:
Remove console.log

Raruto commented:
> it was related to (i don't remember) 

@volterra79 it was added in here: https://github.com/g3w-suite/g3w-client/pull/258

volterra79 commented:
> > it was related to (i don't remember)
> 
> @volterra79 it was added in here: #258

@Raruto it was an issue on g3w-admin code

Raruto commented:
> it was related to (i don't remember)
> > @volterra79 it was added in here: https://github.com/g3w-suite/g3w-client/pull/258
> > > @raruto it was an issue on g3w-admin code

@volterra79 ok, so before merge this one, I would let @leolami test https://github.com/g3w-suite/g3w-client/pull/378 in order to confirm that it will not break: https://github.com/g3w-suite/g3w-client/pull/258

volterra79 commented:
> > it was related to (i don't remember)
> > > @volterra79 it was added in here: #258
> > > > @Raruto it was an issue on g3w-admin code
> 
> @volterra79 ok, so before merge this one, I would let @leolami test #378 in order to confirm that it will not break: #258

@Raruto i checked and it works

volterra79 committed:
Merge branch 'dev' into fix_ux_single_categoriey

Raruto committed:
refactor function `CatalogLayerLegend::setLayerCategories()`

- extract function: `_setAllLayerCategories()`
- extract function: `_setLayerCategories()`

Raruto commented on src/components/CatalogLayerLegend.vue:
```diff
@@ -151,56 +147,68 @@
-
         } catch(err) {
           this.setError();
         }
```
be aware, here I've removed a nested `try..catch` (I hope it was superfluous)

Raruto commented on src/components/CatalogLayerLegend.vue:
```diff
@@ -151,56 +147,68 @@
-            } catch(err) {
-              this.setError();
+            const { nodes = [] } = await projectLayer.getLegendGraphic({ all });
+            if (all) { // case of all categories
+              this._setAllLayerCategories(nodes);
+            } else {
+              this._setLayerCategories(nodes, categories);
```
Can you find a more meaningful name for the two functions: `_setAllLayerCategories()` and `_setAllLayerCategories()` (for me it's not so not clear what the difference between the two `if-else` conditions)

volterra79 replied:
@Raruto done

Raruto committed:
brackets

volterra79 committed:
Merge branch 'dev' into fix_ux_single_categoriey

volterra79 committed:
Rename metho _setLayerCategories to _updateLayerCategories
